src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeAtClauseBlockCheck.java

```diff
@@ -0,0 +1,184 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks that one blank line before the at-clauses block if it is present in Javadoc.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">
+ * Tight-HTML Rules</a>.
+ * Type is {@code boolean}. Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;RequireEmptyLineBeforeAtClauseBlock&quot;/&gt;
+ * </pre>
+ * <p>
+ * By default, the check will report a violation if there is no blank line before the at-clause.
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * testMethod's javadoc.
+ *  * &#64;return something (violation)
+ *  *&#47;
+ * public boolean testMethod() {
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
```

Comment:
If something force you to reference all messages ?

Reply:
Other files do same

Reply:
RequireEmptyLineBeforeAtClauseBlockCheck extends from AbstractJavadocCheck. As a result, RequireEmptyLineBeforeAtClauseBlockCheck needs to document those inherited properties. Some unit test fails if I don't. I don't remember which test.

I see three approaches to handling this:
1. Move those properties out of JavadocCheck into only the classes that needed it
2. Don't extend from AbstractJavadocCheck
3. Disable that unit test for JavadocCheck